src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+                    LocationComponent location = entity.getComponent(LocationComponent.class);
+                    float height = location.getLocalPosition().getY();
+                    //   float deltaHeight = location.getLastPosition().getY();
+                    // float lastHeight = height - deltaHeight;
+                    if (height > thresholdHeight) {
+                        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+                        entity.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+
+                            EntityRef particleEntity = entityManager.create("climateConditions:snowParticleEffect");
+                            LocationComponent targetLoc = entity.getComponent(LocationComponent.class);
+                            LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+                            childLoc.setWorldPosition(targetLoc.getWorldPosition());
+                            Location.attachChild(entity, particleEntity);
+                            particleEntity.setOwner(entity);
+                            Vector3f direction = targetLoc.getLocalDirection();
+                            direction.normalize();
+
+                            //NOTE: initializing velocity using the constructor wasn't working hence the long method.
+                            //particleEntity.addComponent(new VelocityRangeGeneratorComponent(direction,direction.scale(2)));
+                            VelocityRangeGeneratorComponent velocity = particleEntity.getComponent(VelocityRangeGeneratorComponent.class);
+                            direction.scale((float)0.5);
+                            direction.addY((float)0.5);
+                            velocity.minVelocity = direction;
+                            direction.scale((float)1.5);
+                            velocity.maxVelocity = direction;
+                            particleEntity.addOrSaveComponent(velocity);
+                            if(!entity.hasComponent(SnowParticleComponent.class)) {
+                                entity.addComponent(new SnowParticleComponent());
+                            }
+                            entity.getComponent(SnowParticleComponent.class).particleEntity = particleEntity;
+                    }
+            }
+        }
+    }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL + 10 )
+    public void onHealthRegen(ActivateRegenEvent event, EntityRef player, LocationComponent location) {
+        float height = location.getLocalPosition().getY();
+        if(height > thresholdHeight){
+            player.send(new DeactivateRegenEvent());
+        }
+    }
+    @ReceiveEvent
+    public void OnPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, CharacterMovementComponent movement) {
+        defaultRunFactor = movement.runFactor;;
```

Comment:
I think it could be that a player spawns without having the "default" values there... 🤔 I think there is an event for the first character spawn, and otherwise loading the entity's parent prefab and taking those values might be a better option here.

Reply:
Yes I am aware of that.., I didn't know about parent prefabs though .. Will make the changes.